jsons/_load_impl.py

```diff
@@ -103,14 +103,17 @@ def _do_load(json_obj: object,
              cls: type,
              initial: bool,
              **kwargs):
+    if deserializer is None:
+        raise DeserializationError('No deserializer for type {}'.format(get_class_name(cls)), json_obj, cls)
     try:
         result = deserializer(json_obj, cls, **kwargs)
         validate(result, cls, kwargs['fork_inst'])
     except Exception as err:
         clear()
         if isinstance(err, JsonsError):
             raise
-        raise DeserializationError(str(err), json_obj, cls)
+        message = 'Could not deserialize value {} into {}. {}'.format(json_obj, get_class_name(cls), err)
```

Comment:
I am putting also original error into the new message, because it could be helpful to know why the deserializer failed. In case of my last test the full error message is `"Could not deserialize value every day into datetime. not enough values to unpack (expected 2, got 1)"`